From 50135fe5c11c14c7c24727d199c802020931a55b Mon Sep 17 00:00:00 2001
From: Rui Wang <ruwang@redhat.com>
Date: Thu, 21 Aug 2014 04:13:48 -0400
Subject: [acpi] Clean up inclusions of ACPI header files

Message-id: <1408594432-28006-14-git-send-email-ruwang@redhat.com>
Patchwork-id: 88488
O-Subject: [PATCH RHEL7.1 BZ1076655 13/17] ACPI: Clean up inclusions of ACPI header files
Bugzilla: 1076655
RH-Acked-by: Prarit Bhargava <prarit@redhat.com>
RH-Acked-by: Tony Camuso <tcamuso@redhat.com>

From: "Chen, Gong" <gong.chen@linux.intel.com>

Backport of:
Original commit: 8b48463f89429af408ff695244dc627e1acff4f7
Original Author: Lv Zheng <lv.zheng@intel.com>
Original Date: Tue, 3 Dec 2013 08:49:16 +0800

 Replace direct inclusions of <acpi/acpi.h>, <acpi/acpi_bus.h> and
 <acpi/acpi_drivers.h>, which are incorrect, with <linux/acpi.h>
 inclusions and remove some inclusions of those files that aren't
 necessary.

 First of all, <acpi/acpi.h>, <acpi/acpi_bus.h> and <acpi/acpi_drivers.h>
 should not be included directly from any files that are built for
 CONFIG_ACPI unset, because that generally leads to build warnings about
 undefined symbols in !CONFIG_ACPI builds.  For CONFIG_ACPI set,
 <linux/acpi.h> includes those files and for CONFIG_ACPI unset it
 provides stub ACPI symbols to be used in that case.

 Second, there are ordering dependencies between those files that always
 have to be met.  Namely, it is required that <acpi/acpi_bus.h> be included
 prior to <acpi/acpi_drivers.h> so that the acpi_pci_root declarations the
 latter depends on are always there.  And <acpi/acpi.h> which provides
 basic ACPICA type declarations should always be included prior to any other
 ACPI headers in CONFIG_ACPI builds.  That also is taken care of including
 <linux/acpi.h> as appropriate.

Signed-off-by: Chen, Gong <gong.chen@linux.intel.com>
Signed-off-by: Jarod Wilson <jarod@redhat.com>

diff --git a/drivers/acpi/acpi_extlog.c b/drivers/acpi/acpi_extlog.c
index a6869e1..2635a01 100644
--- a/drivers/acpi/acpi_extlog.c
+++ b/drivers/acpi/acpi_extlog.c
@@ -9,7 +9,6 @@
 
 #include <linux/module.h>
 #include <linux/acpi.h>
-#include <acpi/acpi_bus.h>
 #include <linux/cper.h>
 #include <linux/ratelimit.h>
 #include <asm/cpu.h>
-- 
1.7.1